clang/lib/CodeGen/CodeGenAction.cpp

```diff
@@ -1070,14 +1071,23 @@ CodeGenAction::CreateASTConsumer(CompilerInstance &CI, StringRef InFile) {
     CI.getPreprocessor().addPPCallbacks(std::move(Callbacks));
   }
 
+  std::vector<std::unique_ptr<ASTConsumer>> AdditionalConsumers;
+  AdditionalConsumers.reserve(2);
+
   if (CI.getFrontendOpts().GenReducedBMI &&
       !CI.getFrontendOpts().ModuleOutputPath.empty()) {
-    std::vector<std::unique_ptr<ASTConsumer>> Consumers(2);
-    Consumers[0] = std::make_unique<ReducedBMIGenerator>(
+
+    AdditionalConsumers.push_back(std::make_unique<ReducedBMIGenerator>(
         CI.getPreprocessor(), CI.getModuleCache(),
-        CI.getFrontendOpts().ModuleOutputPath);
-    Consumers[1] = std::move(Result);
-    return std::make_unique<MultiplexConsumer>(std::move(Consumers));
+        CI.getFrontendOpts().ModuleOutputPath));
+  }
+
+  if (CI.getFrontendOpts().ClangIRAnalysisOnlyPipeline)
```

Comment:
You need to isolate this addition under cmake generated CIR macro, because we can build clang without CIR support, there are examples around.

Reply:
Done